applications/MappingApplication/custom_utilities/nearest_element_mapper.h

```diff
@@ -105,47 +105,77 @@ class NearestElementMapper : public Mapper
     /* This function maps from Origin to Destination */
     void Map(const Variable<double>& rOriginVariable,
              const Variable<double>& rDestinationVariable,
-             Kratos::Flags Options) override
+             Kratos::Flags MappingOptions) override
     {
         double factor = 1.0f;
 
-        if (Options.Is(MapperFlags::CONSERVATIVE))
+        if (MappingOptions.Is(MapperFlags::CONSERVATIVE))
         {
             factor = MapperUtilities::ComputeConservativeFactor(
                          mNumNodesOrigin,
                          mNumNodesDestination);
         }
 
-        mpMapperCommunicator->TransferInterpolatedData(rOriginVariable,
-                rDestinationVariable,
-                Options,
-                factor);
+        ProcessMappingOptions(MappingOptions, factor);
+
+        // Creating the function pointers for the InterfaceObjects
+        auto function_pointer_origin = std::bind(&GetInterpolatedValueFromGeometryScalar, 
+                                      std::placeholders::_1, 
+                                      rOriginVariable,  
+                                      MappingOptions, 
+                                      std::placeholders::_2);
+
+        auto function_pointer_destination = std::bind(&SetValueOfNode<double>, 
+                                              std::placeholders::_1, 
+                                              std::placeholders::_2,
+                                              rDestinationVariable,  
+                                              MappingOptions,
+                                              factor);
+
```

Comment:
I see that we have these few last lines (121--134) repeated at several points. Maybe, as we start getting other types of operations to use as function_pointer_... we'll want to move them to some kind of factory.